refactor: adjust hash for metadata

preschian opened the pull request:
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Context

- [x] Related: https://github.com/kodadot/ops-internal/issues/1659
- [x] Need workers deployment: https://github.com/kodadot/private-workers/pull/165

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

hassnian reviewed (commented):
1. `PaidMint.vue` missing `close` emit
2. `HolderOfGenerative.vue` needs to be alsos updated like `PaidGenerative.vue` did

prury commented:
what about this one?

preschian commented:
Oh, sorry, I missed the notification about this. will update

preschian commented:
> 1. `PaidMint.vue` missing `close` emit
> 2. `HolderOfGenerative.vue` also needs to be updated like `PaidGenerative.vue` did

Updated description: I slightly changed the PR. 
- I reverted the `PaidMint.vue`; it turns out I got a random reload on a random page, not just a drops page.
- `CollectionDropModalPaidMint` on `HolderOfGenerative.vue` seems like already the same with `PaidGenerative.vue`

hassnian commented on components/collection/drop/PaidGenerative.vue:
```diff
@@ -106,17 +106,8 @@ const handleSubmitMint = async () => {
-
-const handleMintModalClose = () => {
-  closeMintModal()
-  clearMassMint()
```
now we are not clearing the massmint on modal close,

preschian replied:
updated

Jarsen136 commented:
This PR removes the hash while minting the drop. It means the preview image on the drop page would not be the same as the item he minted finally. There should be something else needs to be implemented on the image preview component. Correct me if I'm wrong.

preschian commented:
> This PR removes the hash while minting the drop. It means the preview image on the drop page would not be the same as the item he minted finally. There should be something else needs to be implemented on the image preview component. Correct me if I'm wrong.

Correct 👍🏻 
I think we can simplify the "Mint Drop" modal. It would reduce some of the code, and the user would not need to wait for the preview to render (especially for massmint).

Jarsen136 commented:
> > This PR removes the hash while minting the drop. It means the preview image on the drop page would not be the same as the item he minted finally. There should be something else needs to be implemented on the image preview component. Correct me if I'm wrong.
> 
> Correct 👍🏻 I think we can simplify the "Mint Drop" modal. It would reduce some of the code, and the user would not need to wait for the preview to render (especially for massmint).
> 
> I am planning to do that in a separate branch (branching from this: #10336)

Great! Let's leave it to separate PR.

prury commented:
i understand that we still got to adjust the modal flow after that change, but the nft i minted using this PR has no image: 

https://deploy-preview-10148--koda-canary.netlify.app/ahk/gallery/379-2405285173

[Screencast from 2024-05-24 11-02-42.webm](https://github.com/kodadot/nft-gallery/assets/36627808/f71d77d1-be0c-4e71-888a-8ebf809e0e95)

prury commented on locales/en.json:
```diff
@@ -1598,7 +1598,7 @@
     "signTransaction": "Sign Transaction",
     "paidDropWhyTooltip": "A balance above <strong>{0}</strong> is required to maintain your account's activity, covering the <strong>{1} Existential Deposit,</strong> and to enable seamless future actions.",
-    "createNewVariation": "Create your new variation",
+    "createNewVariation": "Preview Iteration",
```
```suggestion
    "createNewVariation": "Preview Variations",
```

prury commented:
i don't think its working properly @preschian 

[Screencast from 2024-05-27 11-03-20.webm](https://github.com/kodadot/nft-gallery/assets/36627808/69d8199c-966b-415f-9187-e37a51efcf1e)

its generating images without sn or proper variation: 

![image](https://github.com/kodadot/nft-gallery/assets/36627808/10801f5d-66a7-4084-b721-be14a37484a3)

preschian commented:
> its generating images without sn or proper variation:
> (the one that worked is from canary)

Need to wait it for a moment. Currently, the blockchain itself is unable to mint and set metadata at the same time. Currently, we solve it by updating the metadata (image preview, name, sn, animationUrl, etc.) synchronously. refer: https://github.com/kodadot/private-workers/issues/87#issuecomment-2006716343

<img width="1410" alt="Screenshot 2024-05-27 at 21 18 33" src="https://github.com/kodadot/nft-gallery/assets/734428/0b13b591-d1d4-4f87-9534-cf7b8f21f93d">

prury commented:
> > its generating images without sn or proper variation:
> > (the one that worked is from canary)
> 
> Need to wait it for a moment. Currently, the blockchain itself is unable to mint and set metadata at the same time. Currently, we solve it by updating the metadata (image preview, name, sn, animationUrl, etc.) synchronously. 


oh, cool, ok!
